src/api/app/components/add_review_dropdown_component.html.haml

```diff
@@ -10,8 +12,9 @@
     .dropdown-menu.dropdown-menu-start
       %h5.dropdown-header Give a review for...
       - @my_open_reviews.each do |review|
-        = button_tag(type: 'button', class: 'dropdown-item', data: { 'bs-toggle': 'collapse', 'bs-target': '#review-form-collapse',
-                    review: review.id }, aria: { expanded: 'false', controls: 'review-form-collapse' }) do
+        = button_tag(type: 'button', class: 'dropdown-item', aria: { expanded: 'false', controls: 'review-form-collapse' },
+                     data: { 'bs-toggle': 'collapse', 'bs-target': '#review-form-collapse', review: review.id,
+                             'review-reason': helpers.render_as_markdown(review.reason_when_review_was_requested) }) do
```

Comment:
This is generating N+1 queries, fetching one `HistoryElement::RequestReviewAdded` for every review. To fix this, we need to change the request#show action.

Reply:
True. The approach of this PR could introduce more disadvantages than benefits to the user experience. This needs a rethingking. Closing.